CONTRIBUTING.md

```diff
@@ -0,0 +1,256 @@
+
+### Testing
+
+Prior to submitting your code, ensure it is thoroughly tested, both manually and with automated test suites. The
+maintenance of high-quality code is paramount to ensure the longevity and functionality of our project.
+
+Testing of MySQL is a whole big thing, so please go to
+the [ official documentation on MySQL testing ](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_TESTING_TOOLS.html)
+first.
+
+If it is bug fix, make sure that corresponding tests from the test suite work fine. It's a good idea to patch existing
+tests to cover your specific case. For the introduction of new features, the development of a comprehensive test suite
+is strongly recommended.
+
+## Preparing the Pool Request
+
+After finishing the tasks outlined above, you must prepare your branch for a pull request.   
+Please do not forget to run ``` git clang-format ``` on your code before pushing to the GitHub repository.
+
+Then, squash all commits to one or a couple of logically consistent commits.
+For example, the fix of bug should have just one commit. If you have a new feature that requires some changes to the
+existing code, then the first commit should contain required changes, and the second is a new feature and tests. It is
+not a dogma, on the other hand, we value a clean and consistent Git history. Please help us maintain this standard.
+
+Now you can push your local branch to your GitHub repository and make the pull request to the Percona repository.
+
+Please, write meaningful title for the PR and put the number of Jira issue at the beginning of the first line in the
+description. The second line is usually a link to the Jira issue. Then the short description follows. No need to write a
+long description because you wrote one in the Jira issue.
+
+Example of a pool request form:
+
+Title:
+```
+PS-9876 Fix of the compression bug
+```
+Descriptopn:
+```
+PS-9876 Fix of the compression bug
+
+https://perconadev.atlassian.net/browse/PS-9876
+
+The first commit is update of zlib, the second is the fixes of compression function calls.
+```
+
+### Peer review
+
+Peer review of the code always goes before PR merge. So, you have to add some core development team members in the "Reviewers" field of PR. Best guess is to assign the PR to Yura Sorokin, our team lead.
+
+Our GitHub pipeline runs a lot of checks on each PR. Please make sure you fixed all comments provided by checks. Well,
+sometime not all of them :) , you understand...
+
+Following this step, the development team will conduct a thorough review of your code at their earliest convenience. Our
+primary objective is to ensure the highest quality standards. We encourage you to interpret our comments as
+opportunities for improvement and constructive feedback.
+
+When the review is finished and all the issues are addressed, your pull request will be approved and merged into the main source tree.
+
+Congratulations! You are now a recognized contributor.
+
+## Your Code and MySQL Versions
```

Comment:
Should be moved to the beginning. It's a little late to talk about versioning.

Reply:
Fixed